### PR TITLE
sets up /api/topics endpoint in app.js with TDD, MVC and error handling

### PR DESCRIPTION
**Pull request by EliR94**
hey, here is the completed code for 02-get-topics

It includes:
- set up of the app.js file with the new server
- sets up the test suite
- tests for /api/topics endpoint - happy and sad paths
- set up of topics MVC structure
- code within MVC to get the data from the correct database table back to the client
- error handling for /api/topics endpoint

### CODE REVIEW

File: __tests__/integration.test.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            .expect(200)
+            .then(({body}) => {
+                expect(body.length).toBe(3)
+                body.forEach((topic) => {
+                    expect(topic).toMatchObject({
+                        slug: expect.any(String),
+                        description: expect.any(String)
+                    })
+                })
+            })
```
Comment (Harrison365) [inline __tests__/integration.test.js:37]: Very good tests 👍

File: models/topics.models.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const db = require("../db/connection.js")
+
+exports.fetchTopics = () => {
+    return db
+        .query("SELECT * FROM topics;")
+        .then(({rows}) => rows)
+}
```
Comment (Harrison365) [inline models/topics.models.js:7]: Excellent. Really well done 👍

File: __tests__/integration.test.js (hunk 1 above)
Comment (EliR94) [reply __tests__/integration.test.js:37]: thank you 😄

File: models/topics.models.js (hunk 2 above)
Comment (EliR94) [reply models/topics.models.js:7]: thanks so much 😃

**Pull request merged**